core/tests/exhaustive_sync/trial.rs

```diff
@@ -251,84 +287,126 @@ impl Trial {
+                    if not_shared_by_me_file.id != not_shared_by_me_file.parent {
+                        for sharee_index in 0..self.target_devices_by_user.len() {
+                            if user_index == sharee_index {
+                                continue
+                            }
+                            mutants.push(
+                                self.create_mutation(ShareFile {
+                                    user_index,
+                                    device_index,
+                                    target_user_index: sharee_index,
+                                    name: not_shared_by_me_file.name.clone()
+                                })
+                            );
+                        }
+                    }
                 }
 
-                for folder2 in folders.clone() {
-                    if folder.id != folder.parent {
-                        let folder2_name = if folder2.id == folder2.parent {
-                            "root".to_string()
-                        } else {
-                            folder2.name
-                        };
-                        mutants.push(self.create_mutation(AttemptFolderMove {
-                            client: client_index,
-                            folder_name: parent_name.clone(),
-                            destination_name: folder2_name,
-                        }))
+                for folder in folders.clone() {
+                    let parent_name =
+                        if folder.id == folder.parent { "root".to_string() } else { folder.name };
+
+                    mutants.push(self.create_mutation(NewDocument {
+                        user_index,
+                        device_index,
+                        parent: parent_name.clone(),
+                        name: random_filename(),
+                    }));
+
+                    mutants.push(self.create_mutation(NewMarkdownDocument {
+                        user_index,
+                        device_index,
+                        parent: parent_name.clone(),
+                        name: random_filename() + ".md",
+                    }));
+
+                    mutants.push(self.create_mutation(NewFolder {
+                        user_index,
+                        device_index,
+                        parent: parent_name.clone(),
+                        name: random_filename(),
+                    }));
+
+                    for file in all_files.clone() {
+                        if file.id != file.parent {
+                            mutants.push(self.create_mutation(MoveFile {
+                                user_index,
+                                device_index,
+                                doc_name: file.name.clone(),
+                                destination_name: parent_name.clone(),
+                            }))
+                        }
+                    }
```

Comment:
This replaces the separate branches for moving documents or folders into this folder. The previous folder logic felt a little backwards to me (attempting to move this folder into another folder, whereas with documents it was written as attempting to move a document into this folder) but I'm _pretty sure_ I got this right